modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/ShreddedValidatorEnrichmentSpec.scala

```diff
@@ -0,0 +1,154 @@
+              "connection": {
+                "http": {
+                  "uri": "http://iglucentral.com"
+                }
+              }
+            },
+            {
+              "name": "Iglu Central - GCP Mirror",
+              "priority": 1,
+              "vendorPrefixes": [ "com.snowplowanalytics" ],
+              "connection": {
+                "http": {
+                  "uri": "http://mirror01.iglucentral.com"
+                }
+              }
+            }
+          ]
+        }
+      }
+      """)
+  val processor = Processor("sce-test-suite", "1.0.0")
+  val dateTime = DateTime.now()
+  val blocker: Blocker = Blocker.liftExecutionContext(ExecutionContext.global)
+
+  def processEvents(e: CollectorPayload): IO[Boolean] =
+    EtlPipeline
+      .processEvents[IO](
+        adapterRegistry,
+        enrichmentReg,
+        client.rethrowT.unsafeRunSync(),
+        processor,
+        dateTime,
+        Some(e).validNel
+      )
+      .map(_.forall(p => p.isValid))
+
+  def processEventsDisabled(e: CollectorPayload): IO[Boolean] =
+    EtlPipeline
+      .processEvents[IO](
+        adapterRegistry,
+        disabledEnrichReg,
+        client.rethrowT.unsafeRunSync(),
+        processor,
+        dateTime,
+        Some(e).validNel
+      )
+      .map(_.forall(p => p.isValid))
+
+  val body = SpecHelpers.toSelfDescJson(
+    """[{"tv":"ios-0.1.0","p":"mob","e":"se"}]""",
+    "payload_data"
+  )
+
+  val longStr = "s" * 500
+  val fatBody = SpecHelpers.toSelfDescJson(
+    s"""[{"tv":"$longStr","p":"mob","e":"se" , "tr_ci": "$longStr"}]""",
+    "payload_data"
+  )
+
+  val ApplicationJsonWithCapitalCharset = "application/json; charset=UTF-8"
+  object Snowplow {
+    private val api: (String) => CollectorPayload.Api = version => CollectorPayload.Api("com.snowplowanalytics.snowplow", version)
+    val Tp2 = api("tp2")
+  }
+
+  object Shared {
+    val api = CollectorPayload.Api("com.statusgator", "v1")
+    val source = CollectorPayload.Source("clj-tomcat", "UTF-8", None)
+    val context = CollectorPayload.Context(
+      DateTime.parse("2013-08-29T00:18:48.000+00:00").some,
+      "37.157.33.123".some,
+      None,
+      None,
+      Nil,
+      None
+    )
+  }
+
+  val payload = CollectorPayload(
+    Snowplow.Tp2,
+    SpecHelpers.toNameValuePairs("tv" -> "0", "nuid" -> "123"),
+    ApplicationJsonWithCapitalCharset.some,
+    body.some,
+    Shared.source,
+    Shared.context
+  )
+
+  val fatPayload = CollectorPayload(
+    Snowplow.Tp2,
+    SpecHelpers.toNameValuePairs("tv" -> "0", "nuid" -> "123"),
+    ApplicationJsonWithCapitalCharset.some,
+    fatBody.some,
+    Shared.source,
+    Shared.context
+  )
+//  val oversized = CollectorPayload(api, Nil, None, None, source, context)
+
+  "config should parse" >> {
+    val configJson = json"""{
+      "disable": true
+      }"""
+    val schemaKey = SchemaKey(
+      "com.snowplowanalytics.snowplow.enrichments",
+      "shredded_validator_config",
```

Comment:
I'm struggling to think of a good name for this enrichment, and the best I can think of is `event_schema_validator_config` or `atomic_event_schema_validator_config` -- neither of which I like.

But I don't like `shredded_validator_config` either, and in particular the `shredded` bit.  I assume you called it this because of how the [atomic event iglu schema](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/atomic/jsonschema/1-0-0) doesn't include event/context fields.  But my objections are:

* Enrich emits unshredded data.  In fact Enrich knows nothing about shredding data.
* This enrichment is helpful for people loading non-shredded data to a warehouse (as well as people loading shredded)
* Snowplow's strategy is to really push non-shredded formats going forwards.

Reply:
I agree that `shredded` should not appear, as enrich should not worry about what comes next.
I would simply call it `enriched_validator_config`. I think that we don't need to refer to the schema in the name.